4. simple prompt bot nodejs

benbrown opened the pull request:
For this sample, I started with the structure of the echobot sample, and mixed in @Stevenic's already comprehensive set of prompt samples from here:

https://github.com/Microsoft/botbuilder-js/blob/master/samples/dialogs/prompts-ts/src/app.ts

The end result is a bot that presents a menu of many prompt types, and allows you to loop through them and see them demonstrated.

- [ ] Needs deployment stuff
- [x] I need to check and see if there are other prompt types to demonstrate
- [x] Need to sync with @johnataylor for his .net version

vishwacsena commented on javascript_nodejs/4.simple-prompt-bot/dialogs/mainDialog/index.js:
```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
This sample needs to be simplified. Here's the [spec](https://github.com/Microsoft/BotBuilder-Samples/blob/v4/readme.md) - 
```markdown
Demonstrates prompt pattern by prompting user for a property. Introduces user state .vs. conversation state. Ask for name and prints back those information. 
```
So I'd recommend that we pare back this sample and simplify it to just do a name prompt.

benbrown replied:
Doh! I hadn't seen this document.  Thanks.

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/dialogs/mainDialog/index.js:
```diff
@@ -0,0 +1,197 @@
+
+class MainDialog {
+    /**
+     * 
```
extra line

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/dialogs/mainDialog/index.js:
```diff
@@ -0,0 +1,197 @@
+            }
+        ]);
+
+
```
Extra blank lines.

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/dialogs/mainDialog/index.js:
```diff
@@ -0,0 +1,197 @@
+                await dc.end();
+            }
+        ]);
+
```
Extra blank lines

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/dialogs/mainDialog/index.js:
```diff
@@ -0,0 +1,197 @@
+     * @param {Object} context on turn context object.
+     */
+    async onTurn(context) {
+        // see https://aka.ms/about-bot-activity-message to learn more about the message and other activity types
```
Should this have a "." at the end?

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+const path = require('path');
+const ERROR = 1;
+
+// Import reuqired bot services. See https://ama.ms/bot-services to learn more about the different part of a bot
```
spelling. "required".

cleemullins replied:
Spelling "ama.ms" != "aka.ms"

cleemullins replied:
Grammer: "different part" -> "different parts" (plural).

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+const BOT_FILE = path.join(__dirname, (process.env.botFilePath || ''));
+
+console.log('reading config from ', BOT_FILE);
+// read bot configuration from .bot file. 
```
If you've got a "." at the end, then a capital letter should start the sentence. 

"Read".

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+server.listen(process.env.port || process.env.PORT || 3978, function () {
+    console.log(`\n${server.name} listening to ${server.url}`);
+    console.log(`\nGet Bot Framework Emulator: https://aka.ms/botframework-emulator`);
+    console.log(`\nTo talk to your bot, open simple-prompt-bot.bot file in the Emulator`);
```
Emulator shouldn't be capitalized. 

Should have "." at the end of the sentences.

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+let server = restify.createServer();
+server.listen(process.env.port || process.env.PORT || 3978, function () {
+    console.log(`\n${server.name} listening to ${server.url}`);
+    console.log(`\nGet Bot Framework Emulator: https://aka.ms/botframework-emulator`);
```
grammer. Get the Bot Framework Emulator". (forgot "the")

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+try {
+    botConfig = BotConfiguration.loadSync(BOT_FILE, process.env.botFileSecret);
+} catch (err) {
+    console.log(`Error reading bot file. Please ensure you have valid botFilePath and botFileSecret set for your environment`);
```
"." at the end of the sentence.

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+}
+
+// bot name as defined in .bot file 
+// See https://aka.ms/about-bot-file to learn more about .bot file its use and bot configuration .
```
grammer. Choose either:
[...] "more about .bot files" (plural) 
[...] "more about the .bot file".

cleemullins replied:
extra space before the ".".

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+// Get bot endpoint configuration by service name
+const endpointConfig = botConfig.findServiceByNameOrId(BOT_CONFIGURATION);
+
+// Create adapter. See https://aka.ms/about-bot-adapter to learn more about .bot file its use and bot configuration .
```
same grammer comments

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+    appPassword: endpointConfig.appPassword || process.env.microsoftAppPassword
+});
+
+// Define state store for your bot. See https://aka.ms/about-bot-state to learn more about bots memory service
```
"." at the end. Misc grammer cleanup.

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+});
+
+// Define state store for your bot. See https://aka.ms/about-bot-state to learn more about bots memory service
+// A bot requires a sate store to priciest it dialog and user state between messages
```
spelling. "sate" != "state". 
Grammer. 
"priciest"? Not sure what's intended with that one.

cleemullins replied:
priciest != persist.

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+const memoryStorage = new MemoryStorage();
+// CAUTION: The Memory Storage used here is for local bot debugging only. When the bot
+// is restarted, anything stored in memory will be gone. 
+// For production bots use the Azure CosmosDB storage, Azure Blob, or Azure Table storage provides. 
```
"provides" != "providers".

cleemullins replied:
We no longer support Azure Table Storage. Remove from comment.

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+// is restarted, anything stored in memory will be gone. 
+// For production bots use the Azure CosmosDB storage, Azure Blob, or Azure Table storage provides. 
+// const { CosmosDbStorage } = require('botbuilder-azure');
+// const STORAGE_CONFIGURATION = 'cosmosDB'; // this is the name of the cosmos DB configuration in your .bot file
```
"cosmos DB" should be "CosmosDB".

stevengum replied:
Our references to "CosmosDB" are actually incorrect, we should be using "Cosmos DB" as indicated on [Azure](https://azure.microsoft.com/en-us/services/cosmos-db/).

cleemullins commented on javascript_nodejs/4.simple-prompt-bot/index.js:
```diff
@@ -0,0 +1,83 @@
+// create conversation state with in-memory storage provider. 
+const conversationState = new ConversationState(memoryStorage);
+
+// register conversation state as a middleware. The ConversationState middleware automatically reads and writes conversation sate 
```
Register (capitalize) as you have a "." at the end of the sentence.

cleemullins replied:
"sate" != "state".

cleemullins replied:
"." at the end of the sentence.

benbrown commented:
Hey guys, thanks for the feedback. I think this PR was a little premature as I hadn't finished the readme - apologies for that. On other projects, I have used open PRs to track work rather than signal completeness.

benbrown commented on javascript_nodejs/4.simple-prompt-bot/.env:
```diff
@@ -0,0 +1,2 @@
+botFilePath=simple-prompt-bot.bot
```
👍

benbrown commented on javascript_nodejs/4.simple-prompt-bot/README.md:
```diff
@@ -0,0 +1,39 @@
+Open the simple-prompt-bot.bot file in the Botframework Emulator.
+
+Once connected, the bot will present you with a menu of sample prompts. If it does not, send "menu"
+
```
👍

benbrown commented on javascript_nodejs/4.simple-prompt-bot/package.json:
```diff
@@ -0,0 +1,23 @@
+  "main": "index.js",
+  "scripts": {
+    "start": "node ./index.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
👍

cleemullins commented:
@benbrown Is this PR ready to be merged in?

vishwacsena commented on javascript_nodejs/4.simple-prompt-bot/package.json:
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "console-echobot",
```
this neeeds to be simple-prompt-bot or something to that effect. Most certainly not console-echobot :)

vishwacsena commented on javascript_nodejs/4.simple-prompt-bot/package.json:
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "console-echobot",
+  "version": "1.0.0",
+  "description": "Bot Builder v4 console echobot sample",
```
same here. Need to change this to simple prompt bot.

benbrown commented:
I implemented the suggested changes, and also added some more information to the readme to flesh out the description of the bot, and the scope of the available prompt classes.
